services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/flows/FlowValidationNegativeSpec.groovy

```diff
@@ -0,0 +1,169 @@
+    }
+
+    def getSingleSwitch() {
+        def switches = topology.getActiveSwitches()
+        return [switches, switches].combinations().findAll() { src, dst -> src == dst }.unique().first()
+    }
+
+
+    @Unroll
+    def "Flow validation should fail if #rule switch rule is deleted"() {
+        given: "Flow with a transit switch exists"
+
+        def (Switch srcSwitch, Switch dstSwitch) = getNonNeighbouringSwitches()
+        assert srcSwitch && dstSwitch
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow = northbound.addFlow(flow)
+        assert flow?.id
+        assert Wrappers.wait(20) { "up".equalsIgnoreCase(northbound.getFlow(flow.id).status) }
+        def forward = database.getFlow(flow.id)?.left
+        def involvedSwitches = pathHelper.getInvolvedSwitches(flow.id).collect {it -> it.dpId}
+
+        def newRules = findFlowRules(forward.cookie, involvedSwitches)
+        and: "Rules are installed on all switches"
+        assert newRules.size() > 2
+
+        when: "${rule} switch rule is deleted"
+        def cookie = newRules[newRules.keySet()[item]][0].cookie.toString()
+        northbound.deleteSwitchRules(newRules.keySet()[item], cookie)
+        then: "Flow validation should fail"
+
+        !northbound.validateFlow(flow.id).every {it -> it.asExpected}
+        cleanup: "Delete the flow"
+        if (flow){
+            northbound.deleteFlow(flow.id)
+        }
+        where:
+        rule << ["ingress", "transit", "egress"]
+        item << [0, 1, -1]
+    }
+
+    @Unroll
+    def "Both flow and switch validation should fail if flow rule is missing with #flowconfig flow configuration"() {
+        given: "A ${flowconfig} flow exists"
+        def (src, dest) = method()
+
+        def flow = flowHelper.randomFlow(src, dest)
+        assert flow?.id
+        flow = northbound.addFlow(flow)
+        assert Wrappers.wait(20) { "up".equalsIgnoreCase(northbound.getFlow(flow.id).status) }
+
+        def involvedSwitches = pathHelper.getInvolvedSwitches(flow.id).collect {it -> it.dpId}
+        Flow forward = database.getFlow(flow.id).left
+        def flowRules = findFlowRules(forward.cookie, involvedSwitches)
+
+        when: "A flow rule gets deleted"
+        def cookie = flowRules[flowRules.keySet()[0]][0].cookie.toString()
+        northbound.deleteSwitchRules(flowRules.keySet()[0], cookie)
+
+        then: "Flow and switch validation should fail"
+        !northbound.validateFlow(flow.id).every {it -> it.asExpected}
+        northbound.validateSwitchRules(flowRules.keySet()[0]).missingRules.size() > 0
+        cleanup: "Delete the flow"
+        if (flow) {
+            northbound.deleteFlow(flow.id)
+        }
+        where:
+        flowconfig << ["single switch", "neighbouring", "transit"]
+        method << [{ getSingleSwitch() }, { getNeighbouringSwitches() }, { getNonNeighbouringSwitches() }]
```

Comment:
why all methods are wrapped into closure? The only reason for this would be to delay the method calls until test execution, but I don't really see any reason for that. These methods should run with no issues in `where` block directly